tracer/src/Datadog.Tracer.Native/il_rewriter.cpp

```diff
@@ -661,6 +663,15 @@ HRESULT ILRewriter::Export()
 
             pCurrent = (BYTE*) (pEH + 1);
 
+            // Quote from ECMA-335, I.12.4.2.5 Overview of exception handling:
+            // "The ordering of the exception clauses in the Exception Handler Table is important. If
+            //      handlers are nested, the most deeply nested try blocks shall come before the try blocks that enclose them."
+            // If we will not follow that, we might face InvalidProgramException.
+            // That's why we order the Exception Clauses right before applying them.
+
+            std::sort(m_pEH, m_pEH + m_nEH,
+                  [](EHClause a, EHClause b) { return a.m_pTryBegin->m_offset > b.m_pTryBegin->m_offset && a.m_pTryEnd->m_offset < b.m_pTryEnd->m_offset; });
+
```

Comment:
👏🏻 👏🏻 👏🏻 👏🏻

Reply:
Yeah great catch @GreenMatan 🚀! Really highlight the power of exploration tests.

Reply:
Wow, good stuff!!

Reply:
Thanks! 😊